### PR TITLE
Feature/more useful release task

### PR DESCRIPTION
**Pull request by 1natsu172**
I tried implementing the issue #39.

* Overwrite lightweight tag with annotation tag
* Copy compare URL in clipboard in markdown form
* Catch up to the latest tag with "preversion" script

Please check it when you have time. 🙏

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,9 @@
     "compress": "node tasks/compress.js",
     "pack": "node tasks/pack.js",
     "build": "run-s clean compress pack",
+    "preversion": "git pull --tags",
```
Comment (dideler) [inline package.json]: Consider using `git fetch --tags` which should be equivalent, but maybe less confusing. I think of pull as a fetch and merge, even though it doesn't do that with the `--tags` option, a fetch better describes its behaviour.

File: tasks/annotate-tag.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+const [oldTag, newTag] = shell.exec('git tag', {silent: true}).tail({'-n': 2}).stdout.trim().split('\n');
+
+const compareText = `Changes: https://github.com/dideler/toggle-youtube-comments/compare/${oldTag}...${newTag}`
+
+// overwriting the lightweight tag with an annotated tag.
+shell.exec(`git tag --annotate --force ${newTag} ${newTag}^0 --message "${compareText}"`)
+
+// Copy markdown text.
+clipboardy.writeSync(`---\n\n${compareText}`);
```
Comment (dideler) [inline tasks/annotate-tag.js]: Does this inform the user that something has been copied to their clipboard?

File: tasks/annotate-tag.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const clipboardy = require('clipboardy');
+
+const [oldTag, newTag] = shell.exec('git tag', {silent: true}).tail({'-n': 2}).stdout.trim().split('\n');
+
+const compareText = `Changes: https://github.com/dideler/toggle-youtube-comments/compare/${oldTag}...${newTag}`
+
+// overwriting the lightweight tag with an annotated tag.
+shell.exec(`git tag --annotate --force ${newTag} ${newTag}^0 --message "${compareText}"`)
+
+// Copy markdown text.
```
Comment (dideler) [inline tasks/annotate-tag.js]: What do you think about wrapping these commands in named functions instead of using comments to describe the code?

**Review by dideler — APPROVED** [review]
Thanks for starting this @1natsu172!

Looks good, just a few minor suggestions.

File: package.json (hunk 1 above)
Comment (1natsu172) [reply package.json]: Certainly I feel good with `git fetch --tags`. It is doubtful whether can list the tags of the tracking branch with `git tag` ... I will check once.

File: tasks/annotate-tag.js (hunk 2 above)
Comment (1natsu172) [reply tasks/annotate-tag.js]: Ah, the current situation is not. It is better to display a message on the console. I'll try it.

File: tasks/annotate-tag.js (hunk 3 above)
Comment (1natsu172) [reply tasks/annotate-tag.js]: Well, I don't know until I write it. 🤔 
Once written, if named function wrap seems better, I'll change it and resubmit it!

**Comment by 1natsu172** [comment]
@dideler , Thanks for the suggestion comment.

I'll consider them once!

**Comment by 1natsu172** [comment]
@dideler 

Sorry, I intended to push the fix and left behind. I've included some of your suggestions.

* change git pull to fetch
* display info of copy to clipboard
* wrap named functions

**Comment by dideler** [comment]
Looks great, thanks for another awesome contribution @1natsu172!

**Pull request merged**